src/devices/stepper.rs

```diff
@@ -0,0 +1,526 @@
+    pub fn get_min_failsafe_time(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getMinDataRate(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Get maximum data rate
+    pub fn get_max_failsafe_time(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getMaxDataRate(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Set holding current limit
+    pub fn set_holding_current_limit(&self, holding_current_limit: f64) -> Result<()> {
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_setHoldingCurrentLimit(self.chan, holding_current_limit)
+        })?;
+        Ok(())
+    }
+
+    /// Get holding current limit
+    pub fn get_holding_current_limit(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_getHoldingCurrentLimit(self.chan, &mut value)
+        })?;
+        Ok(value)
+    }
+
+    /// Get is moving
+    pub fn get_is_moving(&self) -> Result<bool> {
+        let mut _v = 0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getIsMoving(self.chan, &mut _v) })?;
+
+        match _v {
+            1 => Ok(true),
+            0 => Ok(false),
+            _ => Err(ReturnCode::UnknownVal),
+        }
+    }
+
+    /// Get position
+    pub fn get_position(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getPosition(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Get minimum position
+    pub fn get_min_position(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getMinPosition(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Get maximum position
+    pub fn get_max_position(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getMaxPosition(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Set rescale factor
+    pub fn set_rescale_factor(&self, rescale_factor: f64) -> Result<()> {
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_setRescaleFactor(self.chan, rescale_factor)
+        })?;
+        Ok(())
+    }
+
+    /// Get rescale factor
+    pub fn get_rescale_factor(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getRescaleFactor(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Set target position
+    pub fn set_target_position(&self, stepper: f64) -> Result<()> {
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_setTargetPosition(self.chan, stepper) })?;
+        Ok(())
+    }
+    /// [NOT IMPLEMENTED] Set target position async TODO
+    pub async fn _set_target_position_async(&self, stepper: f64) -> Result<()> {
+        _ = stepper;
+        unimplemented!();
+    }
```

Comment:
Take this out if it will not be implemented for the release. It would be more confusing to leave it in.

We should probably consider a strategy for `async` interfaces. Normally I prefer totally separate structs/interfaces for sync and async behavior, rather than mixing in one implementation. That seems to be the normal Rust convention, I believe.

Reply:
I'll can leave it commented out as a reminder for later implementation. unless you want me to remove it all together